library/core/src/iter/sources/repeat.rs

```diff
@@ -72,10 +72,32 @@ impl<A: Clone> Iterator for Repeat<A> {
     fn next(&mut self) -> Option<A> {
         Some(self.element.clone())
     }
+
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
         (usize::MAX, None)
     }
+
+    #[inline]
+    fn advance_by(&mut self, n: usize) -> Result<(), usize> {
+        // Advancing an infinite iterator of a single element is a no-op.
+        let _ = n;
+        Ok(())
+    }
+
+    #[inline]
+    fn nth(&mut self, n: usize) -> Option<A> {
+        let _ = n;
+        Some(self.element.clone())
+    }
+
+    fn last(self) -> Option<A> {
+        loop {}
```

Comment:
Wouldn't `self.element` be the last one?
Just optimized from running infinite time to a no-op :D

Reply:
Repeat is an infinite iterator so it has no last element.

Reply:
But it has last element according to `next_back` implementation.

Reply:
Can we please have this behavior instead? I am using it to represent something like `[[*]` in `tr` where it expands however much is required. To make this easier I need `last()` to actually return something.

Is there something to be gain from this being an infinite loop?

Reply:
This code does not change the behavior of repeat when calling `last` from Repeat as it existed prior to this PR. The default implementation it replaces calls `next` until `None` is returned, which never happens for Repeat.